Library/Homebrew/language/node.rb

```diff
@@ -16,6 +16,12 @@ def self.pack_for_installation
       # fed to `npm install` only symlinks are created linking back to that
       # directory, consequently breaking that assumption. We require a tarball
       # because npm install creates a "real" installation when fed a tarball.
+      if File.exist?("package.json")
+        pkg_json = JSON.parse(IO.read("package.json"))
```

Comment:
What happens if the JSON parse fails? Might be worth handling this more gracefully (even if it's just reraising with a better error message).

Reply:
It now puts a message to `stderr` that `package.json` could not be parsed prior to re-raising.